include/swift/RemoteAST/RemoteAST.h

```diff
@@ -211,6 +211,12 @@ class RemoteASTContext {
   /// Given a heap object, resolve its heap metadata.
   Result<remote::RemoteAddress>
   getHeapMetadataForObject(remote::RemoteAddress address);
+
+  /// Given an existential and its static type, resolve its dynamic
+  /// type and address.
```

Comment:
You should document what happens if the value stored in the existential is actually itself an existential.  (I think the most general solution is that only a single step of unwrapping is performed; the caller can pretty easily decide on its own whether to iterate.)